datafusion/optimizer/src/eliminate_limit.rs

```diff
@@ -128,4 +195,44 @@ mod tests {
             \n    TableScan: test";
         assert_optimized_plan_eq(&plan, expected);
     }
+
```

Comment:
By the way, can we add more tests here to handle all combination of `skip` and `fetch` between ancestor and child?

Reply:
yes, I'll add more test cases.